fbpic/particles/spin/inline_functions.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2023, FBPIC contributors
+# Author: Michael J. Quin
+# Scientific supervision: Matteo Tamburini
+# Code optimization: Kristjan Poder
+# License: 3-Clause-BSD-LBNL
+"""
+This file is for use with the Fourier-Bessel Particle-In-Cell code (FB-PIC).
+
+It defines a function to push the 'classical' spin vector, according to the
+Bargmann-Michel-Telegdi equation and other spin-push related functions.
+"""
+import math
+from scipy.constants import c
+
+
+def push_s_BMT(sx_i, sy_i, sz_i, ux_i, uy_i, uz_i, ux_f, uy_f, uz_f,
+               Ex, Ey, Ez, Bx, By, Bz, tauconst, anom):
+    """
+    Push spin of a single particle
+    """
+    # Momentum at a half-timestep forward using Boris assumption
+    ux = (ux_i + ux_f)/2
+    uy = (uy_i + uy_f)/2
+    uz = (uz_i + uz_f)/2
+
+    # Assume 1./gamma at midpoint can be calculated from (average)
+    # momentum at midpoint
+    inv_gamma = 1./math.sqrt( 1. + ux**2 + uy**2 + uz**2 )
+    # Define parameter: 1./(1+gamma)
+    inv_1pgamma = inv_gamma/(1. + inv_gamma)
+    """
+    NOTE! ALGORITHM WRITTEN IN TERMS OF REST FRAME FIELDS.
+    CHECK HOW THE FORM OF BMT EQUATION CHANGES FOR LORENTZ-BOOST.
+    """
```

Comment:
Hm, are you certain that the equations only work in a particular frame?
My guess (without checking explicitly) is that they should work in any frame (i.e. that the equations are invariant under a Lorentz transformation) and that they should therefore automatically work in boosted-frame simulations.
```suggestion
```

(Although the actual initialization of the spin may need to be modified when doing a boosted-frame simulation, so that we get the distribution in the boosted-frame, not the one that the user specifies in the lab frame.)

Reply:
I think you're right, the equations would work in another frame, it's 'just' a matter of transforming the spin components. I will try to look into it a bit more!